### PR TITLE
Documenting upgrade considerations

### PR DESCRIPTION
**Pull request by it33**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -180,6 +199,148 @@ Licensing
...
+
+Prior to upgrading the Mattermost binaries from the previous versions, the below options would need to be manually updated in existing config.json file to migrate successfully. This is a list of changes and their new default values in a fresh install: 
+#### Config.json Changes from v0.7 to v1.0 
+
+##### Service Settings 
+
+- Under `ServiceSettings` in [`config.json`](./config/config.json):
+  - **Moved:** `"SiteName": "Mattermost"` which was added to `TeamSettings`
+  - **Removed:** `"Mode" : "dev"` which deprecates a high level dev mode, now replaced by granular controls
+  - **Renamed:** `"AllowTesting" : false` to `"EnableTesting": false` which allow the use of `/loadtest` slash commands during development
```
Comment (jwilander) [inline CHANGELOG.md]: Should be "allows" I believe

File: CHANGELOG.md [hunk 2]
```diff
@@ -180,6 +199,148 @@ Licensing
...
+  - **Removed:** `"UseSSL": false` boolean replaced by `"ConnectionSecurity": ""` under `Security` with new options: _None_ (`""`), _TLS_ (`"TLS"`) and _StartTLS_ ('"StartTLS"`)
+  - **Renamed**: `"Port": "8065"` to `"ListenAddress": ":8065"` to define address on which to listen. Must be prepended with a colon.
+  - **Removed:** `"Version": "developer"` removed and version information now stored in `model/version.go`
+  - **Removed:** `"Shards": {},` which was not used
```
Comment (jwilander) [inline CHANGELOG.md]: Extra comma after the `{}`

Comment (it33) [inline CHANGELOG.md]: fixed

File: CHANGELOG.md [hunk 3]
```diff
@@ -180,6 +199,148 @@ Licensing
...
+  - This section was removed and settings moved to `FileSettings`
+  - **Renamed and Moved:** `"S3AccessKeyId": ""` renamed `"AmazonS3AccessKeyId": "",` and moved to `FileSettings`
+  - **Renamed and Moved:** `"S3SecretAccessKey": ""` renamed `"AmazonS3SecretAccessKey": "",` and moved to `FileSettings`
+  - **Renamed and Moved:** `"S3Bucket": ""` renamed `"AmazonS3Bucket": "",` and moved to `FileSettings`
+  - **Renamed and Moved:** `"S3Region": ""` renamed `"AmazonS3Region": "",` and moved to `FileSettings`
+
+##### Image Settings 
+
+- Under `Image Settings` in [`config.json`](./config/config.json):
+  - **Renamed:** `"Image Settings"` section to `"FileSettings"`
```
Comment (jwilander) [inline CHANGELOG.md]: There should be no space in `Image Settings`

File: CHANGELOG.md (hunk 1 above)
Comment (it33) [inline CHANGELOG.md]: fixed, thanks,

File: CHANGELOG.md [hunk 4]
```diff
@@ -180,6 +199,148 @@ Licensing
...
+  - **Added:** `"DriverName" : "local"` to specify the file storage method, `amazons3` can also be used to setup S3
+
+##### Email Settings
+
+- Under `Email Settings` in [`config.json`](./config/config.json):
```
Comment (jwilander) [inline CHANGELOG.md]: No space in `EmailSettings`

**Comment by jwilander** [comment]
+1

**Comment by crspeller** [comment]
+1

**Comment by it33** [comment]
Thanks guys!

**Pull request closed without merging**